lib/chef/provider/mount/mount.rb

```diff
@@ -217,6 +217,11 @@ def device_logstring
           end
         end
 
+        def device_sub_space_regex
+          # To escape space character in any device network device name
+          device_mount_regex.gsub(" ", "\\x20")
+        end
+
```

Comment:
Let's rename this `device_with_space_escape` so it's clear what's going on. Let's also add a pretty big comment here like:
```
# For CIFS (and perhaps other remote network mounts) when a space is in the "device name"
# it will appear with the space substituted with a special character. However, when mounting,
# the mount needs to be done with an actual space. This function provides the device name with
# the special character to determine if the device is mounted.
```